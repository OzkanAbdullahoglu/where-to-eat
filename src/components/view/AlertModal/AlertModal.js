/* eslint-disable linebreak-style */
import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { compose } from 'recompose';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mapActions, getIsAlertModalVisible } from '../../../reducers';

export const BareAlertModal = ({
  isAlertModalVisible,
  close,
  warning,
}) => (
  <div>
    <Dialog
      open={isAlertModalVisible}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">{'Input is missing!'}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          {warning}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={close} color="primary">
                        Close
        </Button>
      </DialogActions>
    </Dialog>
  </div>
);

BareAlertModal.propTypes = {
  warning: PropTypes.string,
  isAlertModalVisible: PropTypes.bool,
  close: PropTypes.func,
};

const mapStateToProps = (store) => ({
  isAlertModalVisible: getIsAlertModalVisible(store),
});

const withRedux = connect(mapStateToProps, { ...mapActions });

export default compose(withRedux)(BareAlertModal);
