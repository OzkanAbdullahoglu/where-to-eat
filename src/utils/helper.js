
export const openHoursManage = (obj) => {
  let willOpen = '';
  let willClosed = '';
  let openUntil = '';
  const date = new Date();
  const today = date.getDay();
  const time = parseInt(
    `${JSON.stringify(date.getHours())}${JSON.stringify(date.getMinutes())}`, 10
  );

  if (obj === undefined || obj.open[today] === undefined) {
    return '';
  }
  const closeTimeInt = parseInt(obj.open[today].end, 10);
  const openTime = `${obj.open[today].start.slice(0, 2)}:${obj.open[today].start.slice(2)}`;
  const closeTime = `${obj.open[today].end.slice(0, 2)}:${obj.open[
    today
  ].end.slice(2)}`;
  if (closeTimeInt - time < 100) {
    willClosed = `Close soon ${closeTime}`;
  } else if (time > closeTimeInt || time < closeTimeInt) {
    willOpen = `Opens at ${openTime}`;
  } else {
    openUntil = `Open until ${closeTime}`;
  }
  return (
    willClosed || willOpen || openUntil
  );
};

export const getCategoriesTitle = (arr) => {
  const titleArr = arr.map((obj) => obj.title);
  const concatenateTitles = titleArr.reduce(
    (accTitles, singleTitle) => `${accTitles}, ${singleTitle}`
  );
  return concatenateTitles;
};
