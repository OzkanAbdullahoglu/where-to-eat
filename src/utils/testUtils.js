/* eslint-disable linebreak-style */
import { createStore, applyMiddleware } from 'redux';
import { middlewares } from '../store';
import rootReducer from '../reducers';


export const storeFactory = (initalState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initalState);
}
;
