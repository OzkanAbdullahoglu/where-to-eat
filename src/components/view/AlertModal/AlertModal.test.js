/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AlertModal, { BareAlertModal } from './AlertModal';
import { storeFactory, initialStates } from '../../../utils/testUtils';

describe('AlertModal Component', () => {
  const setupUnconnected = (state = {}, props = {}) => {
    const store = storeFactory(state);
    const wrapper = shallow(<BareAlertModal store={store} {...props} />);
    return wrapper;
  };
  const setup = (state = {}) => {
    const store = storeFactory(state);
    const wrapper = shallow(<AlertModal store={store} />).dive();
    return wrapper;
  };
  const closeMocked = jest.fn();
  const props = {
    isAlertModalVisible: false,
    close: closeMocked,
    warning: 'Test Warning Phrase',
  };
  const AlertModalWrapper = setupUnconnected({}, props);

  it('renders Alert Modal Component', () => {
    expect(toJson(AlertModalWrapper)).toMatchSnapshot();
  });

  describe('Alert Modal Renders Material UI Components', () => {
    it('contains a Dialog Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(Dialog))').exists()).toBe(true);
    });
    it('contains a Dialog Title Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(DialogTitle))').exists()).toBe(true);
    });
    it('contains a Dialog Content Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(DialogContent))').exists()).toBe(true);
    });
    it('contains a Dialog Content Text Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(DialogContentText))').exists()).toBe(true);
    });
    it('contains a Dialog Actions Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(DialogActions))').exists()).toBe(true);
    });
    it('contains a Button Component', () => {
      expect(AlertModalWrapper.find('WithStyles(ForwardRef(Button))').exists()).toBe(true);
    });
  });

  describe('Alert Modal Render Props Correctly', () => {
    it('renders `isAlertModalVisible` props', () => {
      const openProp = AlertModalWrapper.find('WithStyles(ForwardRef(Dialog))').prop('open');
      expect(openProp).toEqual(props.isAlertModalVisible);
    });
    it('renders `warning` props', () => {
      const warningProp = AlertModalWrapper.find('WithStyles(ForwardRef(DialogContentText))').text();
      expect(warningProp).toEqual(props.warning);
    });

    it('renders `close` props', () => {
      const closeProp = AlertModalWrapper.find('WithStyles(ForwardRef(Button))');
      closeProp.simulate('click');
      expect(props.close.mock.calls.length).toBe(1);
    });
  });

  describe('redux properties', () => {
    const AlertModalConnectedWrapper = setup(initialStates, {});
    const isAlertModalVisibleProp = AlertModalConnectedWrapper.props()
      .store.getState().map.toggleAlertModal;
    it('has access to `isAlertModalVisible` state', () => {
      expect(isAlertModalVisibleProp).toEqual(props.isAlertModalVisible);
    });
  });
});
