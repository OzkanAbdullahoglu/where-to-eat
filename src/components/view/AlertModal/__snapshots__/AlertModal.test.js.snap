// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal Component renders Alert Modal Component 1`] = `
<div>
  <WithStyles(ForwardRef(Dialog))
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="alert-dialog-title"
    >
      Input is missing!
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <WithStyles(ForwardRef(DialogContentText))
        id="alert-dialog-description"
      >
        Test Warning Phrase
      </WithStyles(ForwardRef(DialogContentText))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[MockFunction]}
      >
        Close
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
